timemachine/fe/free_energy.py

```diff
@@ -399,45 +402,46 @@ def estimate_free_energy_given_initial_states(
         [[pair_overlap_from_ukln(u_kln) for u_kln in ukln_by_lambda] for ukln_by_lambda in ukln_by_lambda_by_component]
     )
 
-    # generate figures
+    return PairBarResult(
+        all_dGs,
+        all_errs,
+        dG_errs_by_lambda_by_component,
+        overlaps_by_lambda,
+        overlaps_by_lambda_by_component,
+        ukln_by_lambda_by_component,
+    )
+
+
+def make_pair_bar_plots(
+    initial_states: List[InitialState], result: PairBarResult, temperature: float, prefix: str
+) -> PairBarPlots:
     U_names = [type(U_fn).__name__ for U_fn in initial_states[0].potentials]
     lambdas = [s.lamb for s in initial_states]
 
     overlap_detail_png = make_overlap_detail_figure(
         U_names,
-        all_dGs,
-        all_errs,
-        u_kln_by_component_by_lambda,
+        result.all_dGs,
+        result.all_errs,
+        result.u_kln_by_lambda_by_component.swapaxes(0, 1),
```

Comment:
nit: add comment # u_kln_by_lambda_by_component -> u_kln_by_component_by_lambda

Reply:
Added in https://github.com/proteneer/timemachine/pull/961/commits/3630d6e96171c285bf8fea1b0f7f1eda21fcc0ef